src/nccl_ofi_cuda.c

```diff
@@ -25,13 +25,24 @@ void *nccl_net_ofi_cuFlushGPUDirectRDMAWrites = NULL;
 
 #define STRINGIFY(sym) # sym
 
+#ifndef __cplusplus
```

Comment:
This doesn't seem like a win.  Either compile in C++ or don't.  Let's not maintain both.

Reply:
I'll rewrite this to a C++ file shortly, but really looking just to get something minimally viable out the door that we can build atop here -- this is already at 15 commits, if we try to fix the world in one go this will never happen.